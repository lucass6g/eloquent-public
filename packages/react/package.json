{
  "name": "@eloquent/react",
  "version": "1.0.0",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.mjs",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup",
    "clean": "rm -rf dist && rm -rf node_modules"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.1.0",
    "@types/react": "^18.3.3",
    "@types/react-dom": "^18.3.0",
    "react": "^18.3.1",
    "typescript": "^5.4.5"
  }
}
