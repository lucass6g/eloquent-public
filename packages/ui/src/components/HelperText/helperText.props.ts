export interface RootHelperTextProps
  extends React.HTMLAttributes<HTMLDivElement> {}

export interface LabelHelperTextProps extends React.HTMLAttributes<HTMLSpanElement> {}