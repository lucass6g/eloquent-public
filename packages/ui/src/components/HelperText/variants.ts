import { tv } from "tailwind-variants";

export const helperTextVariants = tv({
    slots: {
        root: "flex flex-col gap-2",
        label: "text-sm text-neutral-500"
    }
})