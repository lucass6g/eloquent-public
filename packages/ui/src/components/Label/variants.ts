import { tv } from "tailwind-variants";

export  const labelVariants = tv({
    base: "text-sm text-foreground font-normal text-neutral-500 leading-4 peer-disabled:cursor-not-allowed peer-disabled:opacity-70",
})