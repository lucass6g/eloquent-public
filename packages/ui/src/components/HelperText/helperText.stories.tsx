import { Meta, StoryObj } from '@storybook/react';
import { HelperText } from '.';

const meta = {
component: () => {
    return (
        <HelperText.Root>
            <HelperText.Label>Seu texto de apoio.</HelperText.Label>
        </HelperText.Root>
    )
},
  title: 'HelperText',
  parameters: {
    layout: 'centered'
},
} satisfies Meta<{}>;

export default meta;
type Story = StoryObj<typeof meta>;

export const HelperTextDefault: Story = {
name: 'HelperText',
args: {},
render: () => {
    return (
        <HelperText.Root>
            <HelperText.Label>Seu texto de apoio.</HelperText.Label>
        </HelperText.Root>
    )
  },
};