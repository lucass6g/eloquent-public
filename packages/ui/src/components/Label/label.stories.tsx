import { Meta, StoryObj } from '@storybook/react/*';
import { Label } from '.';

const meta = {
    title: 'Label',
    argTypes: {
        asChild: {
            description: "Altere o elemento renderizado padrão para aquele passado como filho, mesclando seus adereços e comportamento.",
            type: { name: "boolean" },
        },
        htmlFor: {
            description: "O id do elemento ao qual o rótulo está associado.",
            type: { name: "string"},
        }
      },
    parameters: {
        docs: {
          description: {
            component: 'Aplicado como rótulo em páginas e componentes.'
          },
        },
        layout: 'centered',
      },
    component: Label,
    tags: ['autodocs'],
} satisfies Meta<{}>

export default meta

type Story = StoryObj<typeof meta>

export const Default: Story = {
    name: "Default Label",
    render: () => (
        <Label>Nome da Página</Label>
    )
}