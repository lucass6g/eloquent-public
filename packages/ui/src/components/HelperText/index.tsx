import React from "react";
import { cn } from "@eloquent/styles";
import { LabelHelperTextProps, RootHelperTextProps } from "./helperText.props";
import { helperTextVariants } from "./variants";

const {root, label} = helperTextVariants()

const RootHelperText = React.forwardRef<HTMLDivElement, RootHelperTextProps>(
    function RootHelperText({className, children, ...props}, ref){
        return(
            <div className={cn(root(), className)} ref={ref} {...props}>
                {children}
            </div>
        )
    }
)

const LabelHelperText = React.forwardRef<HTMLSpanElement, LabelHelperTextProps>(
    function LabelHelperText({className, children, ...props}, ref){
        return(
            <span className={cn(label(), className)} ref={ref} {...props}>
                {children}
            </span>
        )
    }
)

export const HelperText = {
    Root: RootHelperText,
    Label: LabelHelperText
}