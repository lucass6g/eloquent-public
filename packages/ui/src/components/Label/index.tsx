"use client"

import * as React from "react"
import * as LabelPrimitive from "@radix-ui/react-label"
import { type VariantProps } from "class-variance-authority"

import { cn } from "@eloquent/styles"
import  { labelVariants } from "./variants"

const Label = React.forwardRef<
  React.ElementRef<typeof LabelPrimitive.Root>,
  React.ComponentPropsWithoutRef<typeof LabelPrimitive.Root> &
    VariantProps<typeof labelVariants>
>(
  function Label({className, ...props}, ref){
    return (
    <LabelPrimitive.Root
      ref={ref}
      className={cn(labelVariants(), className)}
      {...props}
    />
    )
  }
)

export { Label }
