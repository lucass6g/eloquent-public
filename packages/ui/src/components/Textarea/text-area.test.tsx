import { render, screen } from '@testing-library/react';
import { Label } from '../Label';
import { Textarea } from '.';
import { HelperText } from '../HelperText';
import { describe, expect, it } from 'vitest';

const mockTextAreaComponent = (<div className="w-full gap-1.5">
    <Label htmlFor="message-2" className='text-neutral-800 mb-2'>Título</Label>
    <Textarea placeholder="Digite sua mensagem aqui." id="message-2" />
    <HelperText.Label className="text-neutral-500">
        Seu texto de apoio aparecerá aqui.
    </HelperText.Label>
</div>)


describe('Textarea Component', () => {
    it('renders DefaultStory correctly', () => {
        const component = render(mockTextAreaComponent);
        const textarea = screen.getByPlaceholderText('Digite sua mensagem aqui.');
        expect(textarea).toBeInTheDocument();
        expect(component.baseElement).toMatchSnapshot();
    });
    it('renders WithHelper correctly', () => {
        render(mockTextAreaComponent);
        const label = screen.getByText('Título');
        const helperText = screen.getByText('Seu texto de apoio aparecerá aqui.');
        expect(label).toBeInTheDocument();
        expect(helperText).toBeInTheDocument();
    });
});