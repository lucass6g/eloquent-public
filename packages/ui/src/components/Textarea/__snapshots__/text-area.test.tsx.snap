// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Textarea Component > renders DefaultStory correctly 1`] = `
<body>
  <div>
    <div
      class="w-full gap-1.5"
    >
      <label
        class="text-sm font-normal leading-4 peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-neutral-800 mb-2"
        for="message-2"
      >
        Título
      </label>
      <textarea
        class="min-h-[60px] w-full rounded-[6px] border border-input bg-transparent px-3 py-2 text-sm placeholder:text-neutral-500 focus-visible:outline-none focus-visible:border-neutral-400 disabled:cursor-not-allowed disabled:opacity-50"
        id="message-2"
        placeholder="Digite sua mensagem aqui."
      />
      <span
        class="text-sm text-neutral-500"
      >
        Seu texto de apoio aparecerá aqui.
      </span>
    </div>
  </div>
</body>
`;
